### PR TITLE
ARROW-13306: [Java][JDBC] use ResultSetMetaData.getColumnLabel instead of ResultSetMetaData.getColumnName

### PR DESCRIPTION
**Pull request by NinaPeng**
quick fixed using `getColumnLabel` instead of `getColumnName`

### CODE REVIEW

**Comment by liyafan82** [comment]
@NinaPeng Thanks for the PR. Could you please add a test case for the intended behavior?

File: java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcAliasToArrowTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    // insert rows
+    int rowCount = 4;
+    insertRows(rowCount);
+
+    try (ResultSet resultSet = conn.createStatement().executeQuery(QUERY)) {
+      ResultSetMetaData rsmd = resultSet.getMetaData();
+      assertEquals(2, rsmd.getColumnCount());
+
+      // check column name and column label
+      assertEquals(ORIGINAL_COLUMN_NAME, rsmd.getColumnName(1));
```
Comment (NinaPeng) [inline java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcAliasToArrowTest.java:78]: column name is original name in database, but column label is alias in search query

**Comment by NinaPeng** [comment]
> @NinaPeng Thanks for the PR. Could you please add a test case for the intended behavior?

Thanks for your quick response. I have added a test case based on h2 to cover it.

File: java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcAliasToArrowTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.adapter.jdbc.h2;
+
+import static org.junit.Assert.*;
```
Comment (liyafan82) [inline java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcAliasToArrowTest.java:20]: Please unfold this into several single import statements?

File: java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcAliasToArrowTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    conn = DriverManager.getConnection(url);
+    try (Statement stmt = conn.createStatement()) {
+      stmt.executeUpdate(CREATE_STATEMENT);
+    }
+  }
+
+  // This test verifies reading field alias from an H2 database
+  // works as expected.  If this test fails, something is either wrong
+  // with the setup, or the H2 SQL behavior changed.
+  @Test
```
Comment (liyafan82) [inline java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcAliasToArrowTest.java:67]: Please format the comment as standard JavaDoc?

Comment (NinaPeng) [reply java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcAliasToArrowTest.java:70]: Done

File: java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcAliasToArrowTest.java (hunk 2 above)
Comment (NinaPeng) [reply java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcAliasToArrowTest.java]: done

**Comment by NinaPeng** [comment]
@liyafan82 Thanks for your review. anything else need to be fixed? or this pull request can be merged?

**Comment by liyafan82** [comment]
> @liyafan82 Thanks for your review. anything else need to be fixed? or this pull request can be merged?

@NinaPeng The change looks reasonable to me. I want to merge it in a few days, if there are no more comments.

**Comment by NinaPeng** [comment]
> > @liyafan82 Thanks for your review. anything else need to be fixed? or this pull request can be merged?
> 
> @NinaPeng The change looks reasonable to me. I want to merge it in a few days, if there are no more comments.

@liyafan82 got it. thanks

**Comment by liyafan82** [comment]
Merging. Thanks.

**Pull request closed without merging**